aeppl/transforms.py

```diff
@@ -0,0 +1,146 @@
+        phi_inv = cls.backward(value, *inputs)
+        return at.log(at.abs_(jacobian(phi_inv, [value])))
+
+
+@_logprob.register(TransformedRV)
+def trans_uniform_logprob(op, value, *inputs, name=None, **kwargs):
+
+    untrans_value = value.tag.untrans_value
+    logprob = _logprob(op.base_op, untrans_value, *inputs, name=name, **kwargs)
+
+    jacobian = op.log_jac_det(untrans_value, *inputs)
+
+    if name:
+        logprob.name = f"{name}_logprob"
+        jacobian.name = f"{name}_logprob_jac"
+
+    return logprob + jacobian
+
+
+@local_optimizer(tracks=None)
+def transform_logprob(fgraph: FunctionGraph, node: Node) -> Optional[List[Node]]:
+
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+
+    if rv_map_feature is None:
+        return
+
+    if not isinstance(node.op, RandomVariable):
+        return
+
+    trans_node = _transformed_rv(node.op, node)
+
+    if trans_node is not None:
+        rv_var = node.outputs[1]
+        old_value_var = rv_map_feature.rv_values[rv_var]
+        del rv_map_feature.rv_values[rv_var]
+
+        # We now assume that the old value variable represents the *transformed space*.
+        # This means that we need to replace all instance of the old value variable
+        # with "inversely/un-" transformed versions of itself, so that
+        # variables that depend on this random variable will stay consistent,
+        # since they originally assumed that this now transformed random
+        # variable's value var was in the untransformed space (i.e. just a
+        # plain value of said random variable).
+        new_value_var = trans_node.op.backward(old_value_var, *trans_node.inputs)
+        rv_map_feature.rv_values[rv_var] = new_value_var
+
+        new_rv_var = trans_node.outputs[1]
+
+        if old_value_var.name:
+            new_value_var.name = f"{old_value_var.name}_{trans_node.op.trans_name}"
+
+        # The new transformed random variable will depend on the old value
+        # variable, the latter of which is now considered to be in the
+        # transformed space.
+        # Because of this, when constructing the transformed log-probability
+        # term we need to use the old value variable (i.e. the one that's now
+        # assumed to be in the transformed space), and not the new transformed
+        # value variable.
+        new_value_var.tag.untrans_value = old_value_var
+        rv_map_feature.rv_values[new_rv_var] = new_value_var
+
+        return trans_node.outputs
+
+
+@singledispatch
+def _transformed_rv(
+    op: Op,
+    node: Node,
+) -> Optional[TensorVariable]:
+    """Create a graph for a transformed log-probability of a ``RandomVariable``.
+
+    This function dispatches on the type of ``op``, which should be a subclass
+    of ``RandomVariable``.  If you want to implement new transforms for a
+    ``RandomVariable``, register a function on this dispatcher.
+
+    """
+    return None
+
+
+class TransUniformRV(TransformedRV):
```

Comment:
Are we going to need a class for each transformable variable or this is just to illustrate a custom transform case?

Can't we have an `interval transform` and dispatch it to a lists of appropriate RandomVariable types?

Reply:
Yes, this is just an illustration, and we can always generalize when it's useful.